app/services/registration_checker.rb

```diff
@@ -30,6 +30,26 @@ def self.update_registration_allowed!(update_request, competition_info, requesti
     validate_waiting_list_position!
     validate_update_status!
     validate_update_events!
+  rescue Dynamoid::Errors::RecordNotFound
+    # We capture and convert the error so that it can be included in the error array of a bulk update request
+    raise RegistrationError.new(:not_found, ErrorCodes::REGISTRATION_NOT_FOUND)
+  end
+
+  def self.bulk_update_allowed!(bulk_update_request, competition_info, requesting_user)
+    raise RegistrationError.new(:unauthorized, ErrorCodes::USER_INSUFFICIENT_PERMISSIONS) unless
+      competition_info.is_organizer_or_delegate?(requesting_user)
+
+    errors = {}
+    bulk_update_request['requests'].each do |update_request|
+      # rubocop:disable Style/RedundantBegin
```

Comment:
why do you need to disable the rubocop error here?

Reply:
Rubocop wants me to have no `begin` statement and just rescue the error at the bottom of the method - but if I do that:
* I don't have access to the `update_request` for populating the user_id as it is out of the scope of the loop (I think). 
* This means I can't access the user_id to add to the errors hash
* So it seemed easiest to keep the rescue in the scope of the loop by just disabling the rubocop warning

Reply:
but it still would be in the scope of the loop right? I think you can just remove the begin

Reply:
You mean like this? 

```ruby
  def self.bulk_update_allowed!(bulk_update_request, competition_info, requesting_user)
    raise RegistrationError.new(:unauthorized, ErrorCodes::USER_INSUFFICIENT_PERMISSIONS) unless
      competition_info.is_organizer_or_delegate?(requesting_user)

    errors = {}
    bulk_update_request['requests'].each do |update_request|
      update_registration_allowed!(update_request, competition_info, requesting_user)
      rescue RegistrationError => e
        errors[update_request['user_id']] = e.error
      end
    end
    raise BulkUpdateError.new(:unprocessable_entity, errors) unless errors == {}
  end
```

If so, that doesn't work, as the syntax isn't valid (there's an extra `end`)